server/events/output_updater.go

```diff
@@ -0,0 +1,161 @@
+package events
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"github.com/runatlantis/atlantis/server/events/command"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/events/vcs"
+	"github.com/runatlantis/atlantis/server/events/vcs/types"
+	"github.com/runatlantis/atlantis/server/logging"
+	"github.com/runatlantis/atlantis/server/lyft/feature"
+)
+
+type OutputUpdater interface {
+	UpdateOutput(ctx *command.Context, cmd PullCommand, res command.Result)
+}
+
+// proxy that forwards req to configured output updater based on VCSHost type
+// need to add this proxy since output_updater is vcs provider agnostic
+type OutputUpdaterProxy struct {
+	outputUpdater map[models.VCSHostType]OutputUpdater
+}
+
+func (c *OutputUpdaterProxy) UpdateOutput(ctx *command.Context, cmd PullCommand, res command.Result) {
+	c.outputUpdater[ctx.HeadRepo.VCSHost.Type].UpdateOutput(ctx, cmd, res)
+}
+
+func NewOutputUpdaterProxy(pullOutputUpdater OutputUpdater, checksOutputUpdater OutputUpdater, logger logging.Logger, featureAllocator feature.Allocator, enableGithubChecks bool) OutputUpdater {
+	// All vcs host have pull output updater configured by default.
+	ouputUpdater := map[models.VCSHostType]OutputUpdater{
+		models.Github:          pullOutputUpdater,
+		models.Gitlab:          pullOutputUpdater,
+		models.AzureDevops:     pullOutputUpdater,
+		models.BitbucketCloud:  pullOutputUpdater,
+		models.BitbucketServer: pullOutputUpdater,
+	}
+
+	if enableGithubChecks {
```

Comment:
No need for this, we will assume github will always use github checks. We can control old way vs new way through feature flags and then delete the old implementations once gitbhub checks are rolled out.